modules/i3s/src/types.ts

```diff
@@ -772,84 +772,141 @@ export type FieldInfo = {
   label: string;
 };
 
-export type I3SWebScene = {
-  header: I3SWebSceneLayer;
-  layers: I3SWebSceneOperationalLayer[];
+export type ArcGisWebSceneData = {
+  header: ArcGisWebScene;
+  layers: OperationalLayer[];
 };
 
-export type I3SWebSceneLayer = {
-  operationalLayers: I3SWebSceneOperationalLayer[];
-  baseMap: I3SWebSceneBaseMap;
-  ground: I3SWEbSceneGround;
+/**
+ * ArcGis WebScene spec - https://developers.arcgis.com/web-scene-specification/objects/webscene/
+ */
+export type ArcGisWebScene = {
+  /**
+   * @todo add type.
+   * Spec - https://developers.arcgis.com/web-scene-specification/objects/applicationProperties/
+   * Configuration of application and UI elements.
+   */
+  applicationProperties?: any;
+  /**
+   * Operational layers contain business data which are used to make thematic scenes.
+   */
+  operationalLayers: OperationalLayer[];
+  /**
+   * Basemaps give the web scene a geographic context.
+   */
+  baseMap: BaseMap;
+  /**
+   * Ground defines the main surface of the web scene, based on elevation layers.
+   */
+  ground: Ground;
+  /**
+   * An object that defines the characteristics of the vertical coordinate system used by the web scene.
+   */
   heightModelInfo: HeightModelInfo;
+  /**
+   * Root element in the web scene specifying a string value indicating the web scene version.
+   * Valid values: 1.8, 1.9, 1.10, 1.11, 1.12, 1.13, 1.14, 1.15, 1.16, 1.17, 1.18, 1.19, 1.20, 1.21, 1.22, 1.23, 1.24, 1.25, 1.26, 1.27
+   */
   version: string;
+  /**
+   * String value indicating the application which authored the webmap
+   */
   authoringApp: string;
+  /**
+   * String value indicating the authoring App's version number.
+   */
   authoringAppVersion: string;
-  // TODO Add presentation type. This data allows to add multiple viewes for particular web scene.
+  /**
+   * A presentation consists of multiple slides, where each slide is a specific view into the web scene.
+   * Spec - https://developers.arcgis.com/web-scene-specification/objects/presentation/
+   * @todo Add presentation type.
+   */
   presentation: any;
-  initialState: I3SWebSceneInitialState;
+  /**
+   * An object that provides information about the initial environment settings and viewpoint of the web scene.
+   */
+  initialState: InitialState;
+  /**
+   * An object used to specify the spatial reference of the given geometry.
+   */
   spatialReference: SpatialReference;
+  /**
+   * Type: String
+   * Valid values:
+   * global
+   * local
+   */
```

Comment:
the comment content can be declared in the type

Reply:
Done